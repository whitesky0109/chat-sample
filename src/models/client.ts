import { UserPayload } from 'public/store/actions/userAction';
import { RoomPayload } from 'public/store/actions/roomAction';

export interface IMessage {
  roomId?: string;
  type: string;
  message: File | string;
}

export interface StoreState {
  user: UserPayload;
  room: RoomPayload;
}
