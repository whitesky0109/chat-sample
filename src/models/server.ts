export interface IMigration {
  migration_id: string;
  action: string;
  args: any[];
}

export interface IRepository {
  migrations(): IMigration[];
}

export interface IService {
  init(): Promise<any>;
}

export interface IMigrationLog {
  id: number;
  migration_id: string;
  action: string;
  success: boolean;
  error: string;
  timestamp: Date;
}

export interface IMigrationLogs {
  // tslint:disable-next-line: variable-name
  [migration_id: string]: IMigrationLog;
}

export interface IInvite {
  roomId: string;
  users: string[];
}

export interface IMessage {
  userId: string;
  type: string;
  message: ArrayBuffer | string;
  date: Date;
  isMy: boolean;
}

export interface IUser {
  [userId: string]: string | null;
}

export interface IRoomInfo {
  name: string;
  users: IUser;
  creator: string;
}

export interface IRoom {
  [roomId: string]: IRoomInfo;
}
