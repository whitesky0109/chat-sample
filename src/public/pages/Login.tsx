import { connect } from 'react-redux';
import { RouteComponentProps } from 'react-router';
import React, { Component, ChangeEvent, ReactNode, KeyboardEvent } from 'react';

// models
import { StoreState } from 'models/client';

// etc
import { reqUserLogin } from '../controllers/socket';

// image
const logo = require('../static/user.png');

export interface Props extends RouteComponentProps {
  user?: string;
}

export interface States {
  value: string;
}

export class Login extends Component<Props, States> {
  state: States = {
    value: '',
  };

  inputChange = (e: ChangeEvent<HTMLInputElement>) => {
    const { value } = e.target;
    this.setState({ value });
  }

  inputKeyPress = (e:KeyboardEvent<HTMLInputElement>) => {

    if (e.keyCode === 13 && !e.shiftKey) {
      e.preventDefault();
      this.onLogin();
    }
  }

  onLogin = () => {
    const { value } = this.state;

    if (value) {
      reqUserLogin(value);
    }
  }

  componentWillMount() {
    const { user, history } = this.props;

    if (user) {
      history.push('/room');
    }
  }

  shouldComponentUpdate(nextProps: Props) {
    const { history } = this.props;

    if (nextProps.user) {
      history.push('/room');
      return false;
    }

    return true;
  }

  render(): ReactNode {
    const { value } = this.state;

    return <div className="container login">
      <div>
        <div className="first">
          <img src={logo} id="icon" alt="User Icon" />
        </div>

        <div className="login-form">
          <input
            type="text"
            placeholder="Type new your ID"
            value={value}
            onChange={this.inputChange}
            onKeyDown={this.inputKeyPress}
          />
          <button onClick={this.onLogin} >Connect</button>
        </div>
      </div>
    </div>;
  }
}

const mapDispatchToProps = {
};

const mapStateToProps = (state: StoreState) => {
  return {
    user: state.user.user,
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Login);
