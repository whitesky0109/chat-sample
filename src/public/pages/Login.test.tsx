import React from 'react';
import { Provider } from 'react-redux';
import { mount, ReactWrapper } from 'enzyme';
import createMockStore from 'redux-mock-store';

import Login, { Props, States } from './Login';

import { StoreState } from 'models/client';

describe('Login', () => {
  let component: ReactWrapper;
  let loginComp: ReactWrapper;
  const mock: any = jest.fn();
  const mockStore = createMockStore();
  const initState: StoreState = {
    user: {},
    room: {},
  };
  const props: Props = {
    match: mock,
    location: mock,
    history: mock,
  };

  const store = mockStore(initState);

  it('render', () => {
    component = mount(<Provider store={store as any}>
      <Login {...props} />
    </Provider>);
  });

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('find login component', () => {
    loginComp = component.find('Login');
    expect(loginComp).toHaveLength(1);
  });

  it('입력 테스트', () => {
    const input: ReactWrapper = loginComp.find('.login-form>input');
    const inputValue = 'abcdefg';

    expect(input).not.toBeUndefined();
    input.simulate('change', { target: { value: inputValue } });

    expect((loginComp.state() as States).value).toBe(inputValue);
  });

});
