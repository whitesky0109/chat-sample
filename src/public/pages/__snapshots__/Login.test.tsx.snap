// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login matches snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Login)
    history={[MockFunction]}
    location={[MockFunction]}
    match={[MockFunction]}
  >
    <Login
      history={[MockFunction]}
      location={[MockFunction]}
      match={[MockFunction]}
    >
      <div
        className="container login"
      >
        <div>
          <div
            className="first"
          >
            <img
              alt="User Icon"
              id="icon"
              src={
                Object {
                  "default": "test-file-stub",
                }
              }
            />
          </div>
          <div
            className="login-form"
          >
            <input
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="Type new your ID"
              type="text"
              value=""
            />
            <button
              onClick={[Function]}
            >
              Connect
            </button>
          </div>
        </div>
      </div>
    </Login>
  </Connect(Login)>
</Provider>
`;
